### PR TITLE
Add methods and tools to ensure spatial metadata properties for MBTiles databases

### PR DESCRIPTION
**Pull request by thisisaaronland**
Version 1.3 of the [MBTiles spec](https://github.com/mapbox/mbtiles-spec/blob/master/1.3/spec.md) says:

> The metadata table SHOULD contain these four rows: bounds, center, minzoom, maxzoom

This PR adds methods to ensure those metadata properties are assigned when a new database is created and tools to update existing database.

This PR introduces backwards incompatible changes to the `tilepack.TileOutputter` interface:

 * Add a `AssignSpatialMetadata(orb.Bound, maptile.Zoom, maptile.Zoom) error` method.

This PR introduces backwards incompatible changes to the `tilepack.MbtilesReader` interface:

* Add a `Metadata() (*MbtilesMetadata, error)` method.

This PR introduces a new `MbtilesMetadata` for working with MBTiles metadata properties.

The PR updates the `cmd/build` and `cmd/merge` tools to call the `MbtilesMetadata` on their respective tile "outputters". Additionally the `cmd/merge` tool has been updated to derive the updated bounds and min/max zoom levels for the set of MBTiles databases being merged.

This PR updates dependencies to their most recent version and requires Go 1.23.

Aside from the backwards incompatible changes to the interfaces this PR introduces no outward facing changes. All the command line tools work the same way and all the tests pass.

### CODE REVIEW

**Comment by thisisaaronland** [comment]
I can see the checks are failing because the `go.mod` file specifies Go 1.23 and `.github/workflows/go.yml` specifies Go 1.20

I am not wed to Go 1.23 but the requirements should probably be bumped to Go 1.21 since I think previous versions have been EOL-ed.

File: cmd/merge/main.go [hunk 1]
```diff
@@ -39,6 +39,55 @@ func main() {
...
+		if i == 0 {
+			outputBounds = bounds
+		} else {
+			outputBounds = outputBounds.Union(bounds)
+		}
+
+		minZoom, err := metadata.MinZoom()
+
+		if err != nil {
+			log.Fatalf("Unable to min zoom for %s, %v", inputFilename, err)
```
Comment (iandees) [inline cmd/merge/main.go]:
```suggestion
			log.Fatalf("Unable to derive min zoom for %s, %v", inputFilename, err)
```

File: cmd/merge/main.go [hunk 2]
```diff
@@ -39,6 +39,55 @@ func main() {
...
+		}
+
+		maxZoom, err := metadata.MaxZoom()
+
+		if err != nil {
+			log.Fatalf("Unable to max zoom for %s, %v", inputFilename, err)
```
Comment (iandees) [inline cmd/merge/main.go]:
```suggestion
			log.Fatalf("Unable to derive max zoom for %s, %v", inputFilename, err)
```

**Review by iandees — APPROVED** [review]

**Comment by iandees** [comment]
If you can rebase or merge in my change in #28 should allow compiling Go 1.23.

**Comment by thisisaaronland** [comment]
Do you need me to do anything else for this?

**Comment by iandees** [comment]
Nope, sorry didn't see the merge.

**Pull request merged**